Nicer links in the API docs

tschaub opened the pull request:
This shortens `module:ol/foo/Bar~Bar` type names wherever possible to `Bar` where we link to the given type.

Here is an example of what the map options look like after this change:

![image](https://user-images.githubusercontent.com/41094/184241250-7b9c51db-4706-4115-8ef2-0431a206e8fa.png)

And the same options before:

![image](https://user-images.githubusercontent.com/41094/184241302-6f0610c6-f1ae-4f3a-b7cc-2febef8835e8.png)

For people who don't know JSDoc's bespoke `module:foo/Bar~Thing#method` syntax, I think the API docs are much more readable without the extra characters.

We should be including sensible text in the `@link` tags.  When we don't do that, this change shortens the text for the links.

MoonE commented:
I've also thought about getting rid of the `module:` prefix for some time. Only showing the base name makes it look even nicer as the full context isn't needed most of the time.

Is the method return type parsed differently? Because no spaces are added there around `|`

tschaub commented:
> Is the method return type parsed differently? Because no spaces are added there around `|`

I added a commit that gives union types in the return signature a bit more breathing room.  Let me know if you were thinking about going the other way.  Although when complex types end up getting stringified by Catharsis, we get spaces around `|` (and we can't really control this).  So I think space around `|` is the most consistent.

MoonE reviewed (approved):
With spaces it is more legible, great work. :+1:

Theres a few instances where the links are already shortened manually, sometimes lowercased `(\{@link +[^ }]+?)(\w+) +\2(\(\))?\}` -> `$1$2}` if you feel like replacing them.

tschaub commented:
> Theres a few instances where the links are already shortened manually, sometimes lowercased `(\{@link +[^ }]+?)(\w+) +\2(\(\))?\}` -> `$1$2}` if you feel like replacing them.

Those cases revealed that `:` needs to be among the characters that we consider a delimiter for extracting the short name.  I added a commit to cover that and updated the instances that used the short name (and potentially `()`) as the link text.

Thanks for the review.

marcjansen commented:
Very nice, thxanks.
